src/main/java/com/teamcubation/AccionService/application/port/out/StockRepositoryPort.java

```diff
@@ -0,0 +1,13 @@
+package com.teamcubation.AccionService.application.port.out;
+
+import com.teamcubation.AccionService.domain.exception.StockNotFoundException;
+import com.teamcubation.AccionService.domain.model.Stock;
+import java.util.List;
+
+public interface StockRepositoryPort {
+    Stock create(Stock stock);
+    Stock findById(long id) throws StockNotFoundException;
+    List<Stock> getAll();
+    Stock update(Stock stock);
+    void deleteById(long id);
+}
```

Comment:
Asi podemos dejarlo genercido para luego completarlo:

```suggestion
package com.teamcubation.AccionService.application.port.out;


import com.teamcubation.AccionService.domain.model.Stock;
import java.util.List;

public interface StockRepositoryPort {
    Stock create(Stock stock) throws Exception;
    //TODO: Agregar la custom exception cuando este definida
    Stock findById(long id) throws Exception;
    List<Stock> getAll();
    Stock update(Stock stock) throws Exception;
    void deleteById(long id);
}

```